### PR TITLE
Remove `open` and `openSet` from `Connection` as they don’t exist any…

### PR DESCRIPTION
**Pull request by jloveridge**
Update `MongoClientOptions` properties to make them more correct.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
 https://mongoosejs.com/docs/api.html#Connection
 https://mongoosejs.com/docs/connections.html

### CODE REVIEW

File: types/mongoose/index.d.ts [hunk 1]
```diff
@@ -200,51 +201,36 @@ declare module "mongoose" {
      */
     constructor(base: typeof mongoose);
 
-    /**
-     * Opens the connection to MongoDB.
-     * @deprecated open() is deprecated in mongoose >= 4.11.0
-     * @param mongodb://uri or the host to which you are connecting
```
Comment (emmanuelgautier) [inline types/mongoose/index.d.ts]: Why breaking compatibility with older version of mongoose ?

Comment (jloveridge) [reply types/mongoose/index.d.ts]: @emmanuelgautier the `open` method is no longer present in mongoose 5.5. I didn't trace back to see when they removed it, my guess would be 5.0, but it most definitely is not there now. As such it is incorrect to have the types indicate that the method is still available.

**Comment by jessetrinity** [comment]
@horiuchi @lukasz-zak @Alorel @jendrikw @ethanresnick @vologab @jussikinnula @ondratra @alfirin @idandrd @various89 @Fazendaaa @NormanPerrin @danmana @stablio @emmanuelgautier @frontendmonster @mingchen @penumbra1 @orblazer @HughKu @niuware @Vladmel1234 

Since this is a popular package, these changes will not be merged without owner approval. Is there any interest? Thanks!

**Review by emmanuelgautier — APPROVED** [review]

File: types/mongoose/index.d.ts (hunk 1 above)
Comment (Fonger) [reply types/mongoose/index.d.ts]: It was me who removed open() and openSet() api docs from mongoose. @jloveridge is right.
Mongoose made this breaking changes at `5.0.0-rc0 / 2017-12-28`

**Comment by bruun** [comment]
This patch seems to have broken the fixes to `keepAlive` done in https://github.com/DefinitelyTyped/DefinitelyTyped/commit/6471e3065240488bb1ee6e079d64265192a10f7c#diff-ec6e6a45ee09ed93f5d81cb304c5a0b3

Looking at the mongoose documentation, it seems they still treat `keepAlive` as a boolean: https://mongoosejs.com/docs/connections.html#keepAlive

**Comment by Fonger** [comment]
@brrun It extends from @types/mongodb with correct types so it should work

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3fa33b4f74361d7a77e43f1d45b216dc4b8e3786/types/mongodb/index.d.ts#L477-L481

**Comment by bruun** [comment]
@Fonger My bad, I had another dependency keeping @types/mongodb at an older version.

**Pull request merged**